src/main/java/org/arend/core/expr/visitor/NormalizeVisitor.java

```diff
@@ -38,7 +38,7 @@ public Expression visitApp(AppExpression expr, NormalizationMode mode) {
     }
     Collections.reverse(args);
 
-    LamExpression lam = function.cast(LamExpression.class);
+    LamExpression lam = function instanceof LamExpression ? (LamExpression) function : null;
```

Comment:
Why did you do that? That's incorrect.

Reply:
This suppresses a NPE when `function == null`

Reply:
`function` cannot be equal `null` and this code is not equivalent to the original. It is incorrect.

Reply:
Ok, I'll revet this single change. Note that this warning is shown by IntelliJ, so it might be related to the `function = expr.getFunction().accept(this, NormalizationMode.WHNF);` line

Reply:
![image](https://user-images.githubusercontent.com/16398479/76269801-7e573300-6249-11ea-9550-0f624534fa57.png)

^